core/src/main/scala/org/broadinstitute/dsde/rawls/serviceperimeter/ServicePerimeterServiceImpl.scala

```diff
@@ -90,7 +92,26 @@ class ServicePerimeterServiceImpl(dataSource: SlickDataSource,
 
   def overwriteGoogleProjectsInPerimeter(servicePerimeterName: ServicePerimeterName,
                                          dataAccess: DataAccess
-  ): ReadAction[Unit] =
+  ): ReadWriteAction[Unit] = overwriteServicePerimeterInternal(servicePerimeterName, dataAccess).asTry.flatMap {
+    case Failure(failure: MissingGoogleProjectsException) =>
+      logger.info(
+        s"Initial service perimeter update operation failed because some Google projects have been deleted. Deleting associated workspaces in Rawls and trying again. [projects=${failure.projectNumbers}]"
+      )
+      dataAccess.workspaceQuery
+        .deleteByGoogleProjectNumbers(failure.projectNumbers)
```

Comment:
I've also considered introducing a new `WorkspaceState` `MissingGoogleProject` and filtering out workspaces with that state when updating the perimeter. At the moment, I'm leaning towards this approach since it would be confusing for users to still be able to see and perform operations on the workspace that may or may not succeed without a Google project. Deleting the workspace entirely is more destructive, but should result in a better user experience at the end of the day.